pipeline/dags/dag_utils/sources/soliguide.py

```diff
@@ -52,7 +54,12 @@ def search(
             places_data += response_data["places"]
             page_number += 1
 
-            if len(places_data) >= response_data["nbResults"]:
+            if total_number_of_pages is None:
+                total_number_of_pages = ceil(
+                    response_data["nbResults"] / data["options"]["limit"]
+                )
```

Comment:
fyi tu peux faire plus simple sans le package `math` :

```suggestion
                total_number_of_pages =  (response_data["nbResults"] // data["options"]["limit"]) + 1
```

`//` est la division euclidienne (le résultat est le "quotient")

ref : https://docs.python.org/3/library/operator.html#mapping-operators-to-functions

Reply:
Tant qu'on a pas de probleme de memoire on peut faire comme ca. Il faudra repasser a la pagination avec des generateur si ca devient trop "gros".